### ISSUE TITLE
Pylint issues

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbem#1206]
Update 2018-10-07: PR #1409 addressed some more issues and got back to the low water mark described below.

Update 2018-09-28: PR #1405 addressed some issues that had been creeping in lately, but the result is still worse than the low water mark listed below.

Low water mark (With PR #1353, as of 2018-08-23):

```
+-----------------------+------------+
|message id             |occurrences |
+=======================+============+
|fixme                  |283         |
+-----------------------+------------+
|unused-argument        |2           |
+-----------------------+------------+
|using-constant-test    |1           |
+-----------------------+------------+
|too-few-public-methods |1           |
+-----------------------+------------+



************* Module pywbem._listener
W:908, 0: TODO: Describe how the processing threads terminate. (fixme)

************* Module pywbem._nocasedict
W:433, 0: TODO #1062: Could compare hash values for better performance (fixme)

************* Module pywbem._recorder
W:641, 0: TODO: future bypassed code to only ouput name and method if the (fixme)

************* Module pywbem._server
W:135, 0: TODO: Clarify which OpenPegasus versions need root/PGInterOp? (fixme)
W:740, 0: TODO: Figure out how to get version of SFCB (fixme)

************* Module pywbem.cim_http
W:615, 0: TODO #893: Consider the use of default_context() (fixme)
W:949, 0: TODO 4/2018 AM Enable retry logic. For unknown reasons, (fixme)
W:953,20: Using a conditional statement with a constant value (using-constant-test)

************* Module pywbem.cim_obj
W:2058, 0: TODO: Ignore host for CIMObject format (fixme)
W:3582, 0: TODO: Ignore host for CIMObject format (fixme)

************* Module pywbem.cim_operations
W:1465, 0: TODO execute stage of wbem connection if not already executed. (fixme)
W:1698, 0: TODO #919: Further tests on this IRETURN or PARAMVALUE (fixme)

************* Module pywbem.mof_compiler
W:2248, 0: TODO #991: CreateClass should reject if the class already exists (fixme)
W:2352, 0: TODO #990: Also roll back changes to qualifier declarations (fixme)

************* Module pywbem.tupleparse
W:1131, 0: TODO #1044: Clarify if hex support is needed. (fixme)
W:1288, 0: TODO #1044: Clarify if hex support is needed. (fixme)
W:1458, 0: TODO #1044: Clarify if hex support is needed (fixme)
W:1488, 0: TODO #1044: Clarify if hex support is needed (fixme)

************* Module pywbem_mock._wbemconnection_mock
W: 76, 0: TODO: ks Future We have not considered that iq and ico are deprecated in (fixme)
W:310, 0: TODO: ks. FUTURE maybe we should really have a subdict per class but (fixme)
W:483, 0: TODO:ks Future we might be able to use our own MOFWBEMRepository to (fixme)
W:822, 0: TODO:ks FUTURE Consider sorting to perserve order of compile/add. (fixme)
W:842, 0: TODO:ks Future: Possibly sort insts by path order. (fixme)
W:1347, 0: TODO:ks Future Remove dup test since we should be (fixme)
W:1650, 0: TODO: should we use prop name from qualifier in override? (fixme)
W:1684, 0: TODO: match qualifiers in override. (fixme)
W:1686, 0: TODO: test class qualifiers. (fixme)
W:1855, 0: TODO:ks FUTURE implement set... method for instance, qualifier, class (fixme)
W:1992, 0: TODO:ks Future use internal function of repo to create namespace (fixme)
W:2223, 0: TODO:ks Future remove this test for duplicate inst paths since (fixme)
W:2489, 0: TODO:ks FUTURE: Make list from _get_reference_classnames if classes (fixme)
W:2793, 0: TODO:ks Future. Use the timeout along with response delay. Then (fixme)
W:3123, 0: TODO FUTURE: add method to repo that allows privileged users (fixme)



************* Module conftest
W:649, 0: TODO 6/18 AM: Loop above should probably be on elems (fixme)
W:947, 0: TODO 2016/07 AM: Improve the presentation of the difference (fixme)
W:994, 0: TODO 2016/07 AM: Improve the presentation of the diff. (fixme)
W:999, 0: TODO redo as indexed loop to compare all items. (fixme)

************* Module run_cim_operations
W: 90, 0: TODO ks Nov 2016 Create decerators for both this and the OpenPegasus (fixme)
W:225, 0: TODO ks aug 17. FUTURE This should be integrated into logging (fixme)
W:756, 0: TODO Apparently ks 5/16 Pegasus did not implement all properties (fixme)
W:800, 0: TODO Apparently ks 5/16 Pegasus did not implement all properties (fixme)
W:1218, 0: TODO ks 5/16 why does pegasus return this particular err. (fixme)
W:1398, 0: TODO this only returns single instance. (fixme)
W:1468, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:1542, 0: TODO ks 5/30 2016 add tests here (fixme)
W:1591, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:1667, 0: TODO ks 5/30 2016 add tests here (fixme)
W:1705, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:1737, 0: TODO ks This not right. we are going back to associators (fixme)
W:1779, 0: TODO ks 5/30 2016 add tests here (fixme)
W:1831, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:1905, 0: TODO ks 5/30 2016 add tests here (fixme)
W:2377, 0: TODO add get and test for change. (fixme)
W:2547, 0: TODO specalize this one for OpenPegasus (fixme)
W:2561, 0: TODO ks Mar 2017. Returns value 1 rather than zero for some reason (fixme)
W:2567, 0: TODO: Call with empty arrays (fixme)
W:2569, 0: TODO: Call with weird VALUE.REFERENCE child types: (fixme)
W:2596, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:2602, 0: TODO: check return values (fixme)
W:2656, 0: TODO ks dec/16 why returning CIM_Collection (fixme)
W:2657, 0: TODO ks dec/16 why do we not get return when we user resultrole (fixme)
W:2769, 0: TODO: For now, disabled test for class name of associated insts. (fixme)
W:2914, 0: TODO. This is really broken (fixme)
W:2926, 0: TODO: For now, disabled test for class name of referencing insts. (fixme)
W:3123, 0: TODO add these tests (fixme)
W:3139, 0: TODO validate qualifiers in properties. here or elsewhere (fixme)
W:3144, 0: TODO validate parameters in methods (fixme)
W:3187, 0: TODO could we assert some size limit here. Probably Not (fixme)
W:3326, 0: TODO extend for options of Deepinheritance, LocalOnly, (fixme)
W:3890, 0: TODO account for openpegasus only response of returning (fixme)
W:4095, 0: TODO ks 5/16 figure out why we could not get instances of a class (fixme)
W:4213, 0: TODO create an instance write it, get it and test results (fixme)
W:4268, 0: TODO Create a new instance on server and test return using (fixme)
W:4391, 0: TODO change to create function to do the hiearchy search (fixme)
W:5498, 0: TODO confirm that this should be pegasus (fixme)
W:6083, 0: TODO modify this so it is not Pegasus dependent (fixme)
W:6114, 0: TODO: ks Finish this test completely when we add other (fixme)
W:6141, 0: TODO ks 6/16 add more tests including: multiple subscriptions, filters (fixme)
W:6348, 0: TODO Future: modify this so it is not Pegasus dependent (fixme)
W:6425, 0: TODO should we context this sub_mgr (fixme)
W:6631, 0: TODO Pegasus Issue.  Why does this code cause 8 indications (fixme)
W:4523,52: Unused argument 'pull_disabled' (unused-argument)
W:4839,47: Unused argument 'pull_disabled' (unused-argument)

************* Module test_cim_obj
W:1074, 0: XXX: Migrate to pytest (fixme)
W:1100, 0: XXX: Migrate to pytest (fixme)
W:1199, 0: XXX: Migrate to pytest (fixme)
W:1212, 0: TODO 01/18 AM Remove test_all() once PR #955 is merged (is in test hash) (fixme)
W:1213, 0: XXX: Migrate to pytest (fixme)
W:1339, 0: TODO 01/18 AM Reformulate these tests in test hash once PR #955 is merged (fixme)
W:1340, 0: XXX: Migrate to pytest (fixme)
W:1412, 0: XXX: Implement with pytest (fixme)
W:1422, 0: XXX: Implement with pytest (fixme)
W:3414, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:3994, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:4087, 0: XXX: Merge with succeeds (fixme)
W:4687, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:4849, 0: XXX: Merge with succeeds (fixme)
W:4871, 0: XXX: Migrate to pytest (fixme)
W:4873, 0: TODO Add tests for CIMInstance property_list argument (fixme)
W:4922, 0: XXX: Migrate to pytest (fixme)
W:4986, 0: XXX: Migrate to pytest (fixme)
W:5014, 0: XXX: Migrate to pytest (fixme)
W:5179, 0: XXX: Implement with pytest (fixme)
W:5189, 0: XXX: Implement with pytest (fixme)
W:6341, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:6394, 0: XXX: Migrate to pytest (fixme)
W:6420, 0: XXX: Migrate to pytest (fixme)
W:6448, 0: XXX: Migrate to pytest (fixme)
W:6492, 0: XXX: Migrate to pytest (fixme)
W:6988, 0: XXX: Merge with succeeds (fixme)
W:8108, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:8331, 0: XXX: Merge with succeeds (fixme)
W:8347, 0: XXX: Migrate to pytest (fixme)
W:8364, 0: XXX: Migrate to pytest (fixme)
W:8400, 0: XXX: Migrate to pytest (fixme)
W:8540, 0: XXX: Implement with pytest (fixme)
W:8547, 0: XXX: Implement with pytest (fixme)
W:12211, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:12309, 0: XXX: Merge with succeeds (fixme)
W:12521, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:12638, 0: XXX: Merge with succeeds (fixme)
W:12654, 0: XXX: Migrate to pytest (fixme)
W:12670, 0: XXX: Migrate to pytest (fixme)
W:12706, 0: XXX: Migrate to pytest (fixme)
W:12734, 0: XXX: Implement with pytest (fixme)
W:12741, 0: XXX: Implement with pytest (fixme)
W:15655, 0: TODO 01/18 AM Enable test case once char16 produces single quotes (fixme)
W:15784, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:15849, 0: XXX: Merge with succeeds (fixme)
W:15961, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:16015, 0: XXX: Merge with succeeds (fixme)
W:16031, 0: XXX: Migrate to pytest (fixme)
W:16051, 0: XXX: Migrate to pytest (fixme)
W:16089, 0: XXX: Migrate to pytest (fixme)
W:16920, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:17188, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:17265, 0: XXX: Merge with succeeds (fixme)
W:17563, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:17692, 0: XXX: Merge with succeeds (fixme)
W:17708, 0: XXX: Migrate to pytest (fixme)
W:17734, 0: XXX: Migrate to pytest (fixme)
W:17870, 0: XXX: Migrate to pytest (fixme)
W:18031, 0: XXX: Implement with pytest (fixme)
W:18038, 0: XXX: Implement with pytest (fixme)
W:18983, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:19042, 0: XXX: Merge with succeeds (fixme)
W:19065, 0: XXX: Merge with succeeds (fixme)
W:19079, 0: XXX: Merge with succeeds (fixme)
W:19328, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:19448, 0: XXX: Merge with succeeds (fixme)
W:19474, 0: XXX: Migrate to pytest (fixme)
W:19498, 0: XXX: Migrate to pytest (fixme)
W:19523, 0: XXX: Migrate to pytest (fixme)
W:19609, 0: XXX: Implement with pytest (fixme)
W:19616, 0: XXX: Implement with pytest (fixme)
W:20034, 0: XXX: Merge with succeeds (fixme)
W:20607, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:20666, 0: XXX: Merge with succeeds (fixme)
W:21053, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:21147, 0: XXX: Merge with succeeds (fixme)
W:21163, 0: XXX: Migrate to pytest (fixme)
W:21186, 0: XXX: Migrate to pytest (fixme)
W:21263, 0: XXX: Migrate to pytest (fixme)
W:21429, 0: XXX: Implement with pytest (fixme)
W:21436, 0: XXX: Implement with pytest (fixme)
W:25185, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:25244, 0: XXX: Mrg. succeeds (fixme)
W:25636, 0: XXX: Migr. to s.tst.fnc. (fixme)
W:25772, 0: XXX: Merge with succeeds (fixme)
W:25788, 0: XXX: Migrate to pytest (fixme)
W:25810, 0: XXX: Migrate to pytest (fixme)
W:25839, 0: XXX: Migrate to pytest (fixme)
W:25890, 0: TODO ks Dec 16. Add more equality tests. (fixme)
W:25903, 0: XXX: Implement with pytest (fixme)
W:25910, 0: XXX: Implement with pytest (fixme)
W:29192, 0: TODO 01/18 AM Enable once char16 uses single quotes (fixme)
W:29497, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:29554, 0: XXX: Implement with pytest (fixme)
W:29561, 0: XXX: Migrate to pytest (fixme)
W:29639, 0: TODO How should pywbem deal with a single, unnamed, keybinding? (fixme)
W:29838, 0: XXX: Migrate to s.tst.fnc. (fixme)
W:29883, 0: XXX: Merge with succeeds (fixme)
W:30430, 0: XXX: Migrate to s.tst.fnc. (fixme)

************* Module test_cim_types
W:405, 0: TODO: Add testcases for get_local_utcoffset() (fixme)
W:406, 0: TODO: Add testcases for now() (fixme)
W:407, 0: TODO: Add testcases for fromtimestamp() (fixme)
W:821, 0: TODO: Add tests for atomic_to_cim_xml() (fixme)

************* Module test_cim_xml
W:  8, 0: TODO: A bunch of tests are still unimplemented for bits of the (fixme)
W:490, 0: TODO: Implement ValueNull test (fixme)
W:1269, 0: TODO: (fixme)
W:1356, 0: TODO: RESPONSEDESTINATION (fixme)
W:1389, 0: TODO: RESPONSEDESTINATION (fixme)
W:1499, 0: TODO: QUALIFIER.DECLARATION (fixme)
W:1587, 0: TODO: SIMPLEREQACK (fixme)
W:1713, 0: TODO: INSTANCE* (fixme)
W:1735, 0: TODO: PARAMTYPE (fixme)
W:1795, 0: TODO: QUALIFIER.DECLARATION (fixme)

************* Module test_itermethods
W: 35, 0: TODO: ks Oct 17 Remove this pylint disable in the future when pylint or (fixme)
W:1627, 0: TODO ks: This assert disabled. (fixme)

************* Module test_logging
W:100, 0: TODO compare detail_level (fixme)
W:378, 0: TODO add test for when connection param exists (fixme)

************* Module test_mof_compiler
W:442, 0: TODO The following is cim version dependent. (fixme)
W:473, 0: TODO ks 4/16 add error test for something not found (fixme)
W:530, 0: TODO ks 6/16why does this generate end-of-file rather than more (fixme)
W:569, 0: TODO ks apr 2016 Grow the number of functions to test property (fixme)
W:751, 0: TODO add test for array value in inst, not scalar (fixme)
W:780, 0: TODO: Apply issues #203, #205 to flavor parms. (fixme)
W:789, 0: TODO: Apply issues #203, #205 to flavor parms. (fixme)
W:824, 0: TODO: Apply issues #203, #205 to flavor parms. (fixme)
W:832, 0: TODO: Apply issues #203, #205 to flavor parms. (fixme)

************* Module test_recorder
W: 91, 0: TODO python 2.7 will not write the following unicode character (fixme)
W:239, 0: TODO add test for reference properties (fixme)
W:240, 0: TODO add test for embedded object property (fixme)
W:299, 0: TODO host does not appear in output yaml. Is that correct??? (fixme)
W:385, 0: TODO fix the following. Currently it fails with: (fixme)
W:730, 0: TODO AM 2018-06: Suppress the printing to stderr (fixme)
W:766, 0: TODO AM 2018-06: Suppress the printing to stderr (fixme)
W:805, 0: TODO AM 2018-06: Suppress the printing to stderr (fixme)
W:1720, 0: TODO add tests for all for invoke method. (fixme)

************* Module test_subscriptionmanager
W: 79, 0: TODO AM 8/18 The inst_from_classname() method is not used. (fixme)
W:294, 0: TODO: ks Finish this test completely when we add other (fixme)

************* Module test_tupleparse
W:6946, 0: TODO 1/18 AM: Should this not be rejected as invalid type? (fixme)
W:9849, 0: TODO: IPARAMVALUE tests (fixme)
W:9850, 0: TODO: EXPPARAMVALUE tests (fixme)
W:9851, 0: TODO: RETURNVALUE tests (fixme)
W:9852, 0: TODO: IRETURNVALUE tests (fixme)
W:9853, 0: TODO: IMETHODCALL tests (fixme)
W:9854, 0: TODO: IMETHODRESPONSE tests (fixme)
W:9855, 0: TODO: METHODCALL tests (fixme)
W:9856, 0: TODO: METHODRESPONSE tests (fixme)
W:9857, 0: TODO: EXPMETHODCALL tests (fixme)
W:9858, 0: TODO: EXPMETHODRESPONSE tests (fixme)
W:9859, 0: TODO: MESSAGE tests (fixme)
W:9860, 0: TODO: SIMPLEREQ tests (fixme)
W:9861, 0: TODO: SIMPLERSP tests (fixme)
W:9862, 0: TODO: SIMPLEEXPREQ tests (fixme)
W:9863, 0: TODO: SIMPLEEXPRSP tests (fixme)

************* Module test_tupletree
W:166, 0: TODO: Cannot handle comments, cdata, processing instructions, etc. (fixme)

************* Module test_wbemconnection_mock
W:913, 0: TODO expand test to use lo parameterized.  Note that (fixme)
W:1133, 0: TODO: Add test of format of the repo outut with a very simple schema (fixme)
W:1284, 0: TODO: Remove alias once inst without alias properly generates path (fixme)
W:1439, 0: TODO test returned instance for match with what was created (fixme)
W:1604, 0: TODO ks confirm whether following needed (fixme)
W:2085, 0: TODO add to test to confirm that any property in superclasses (fixme)
W:2167, 0: TODO test returned instance. Right now only comparing (fixme)
W:2193, 0: TODO extend to test lo and iq (fixme)
W:2204, 0: TODO add tests for iq and ico (fixme)
W:2223, 0: TODO extend to test lo and iq (fixme)
W:2234, 0: TODO test complete instances returned. Now only testing paths (fixme)
W:2344, 0: TODO integrate this into normal test (fixme)
W:2383, 0: TODO change to use algorithm to get list of possible (fixme)
W:2407, 0: TODO compute number expected be in returned. This is number with (fixme)
W:2450, 0: TODO compute number should be returned. Right now fixed number (fixme)
W:2467, 0: TODO improve algorithm so we define exactly what properties (fixme)
W:2487, 0: TODO Test actual instance returned (fixme)
W:2528, 0: TODO Test actual instance returned (fixme)
W:2536, 0: TODO repeat pl test with conn rather than connlite (fixme)
W:2560, 0: TODO Test actual instance returned (fixme)
W:2597, 0: TODO: Still have issue with one test. and am therefore (fixme)
W:2611, 0: TODO Test actual instance returned (fixme)
W:2613, 0: TODO test for ico and iq (fixme)
W:2614, 0: TODO test for pl where repository contains classes (fixme)
W:2615, 0: TODO test for instances do not exist. (fixme)
W:2618, 0: TODO integrate this into normal test (fixme)
W:2839, 0: TODO additional tests. (fixme)
W:3113, 0: TODO fix common test for path equality (fixme)
W:3135, 0: TODO test actual instances (fixme)
W:3357, 0: TODO test instances returned rather than just paths and (fixme)
W:3386, 0: TODO FUTURE: test with timer (fixme)
W:3726, 0: TODO expand test exp_rslt definition. Need to be able to (fixme)
W:3744, 0: TODO enable the above tests (fixme)
W:3770, 0: TODO redo this as a clean compare test (fixme)
W:3787, 0: TODO not sure we really need this testcase? Combine with next case (fixme)
W:3807, 0: TODO test for specific instance. (fixme)
W:3880, 0: TODO test for references propertylist. (fixme)
W:3900, 0: TODO: ks Review all these tests and results. I am uncomfortable (fixme)
W:3979, 0: TODO add more tests (fixme)
W:4026, 0: TODO expand associator instance names tests (fixme)
W:4053, 0: TODO add more tests (fixme)
W:4102, 0: TODO expand associator instance tests (fixme)
W:4116, 0: TODO: Review this and its results. I am uncomfortable I have not (fixme)
W:4138, 0: TODO test error.  Failed if rc was TST_PERSON (fixme)
W:4172, 0: TODO expand associator classes test to test for correct properties (fixme)
W:4239, 0: TODO extend so generates whatever exception defined (fixme)
W:4556, 0: TODO ks Rewrite the following tests into a single test with parameterization (fixme)

************* Module test_wbemserverclass
W:136, 0: TODO Break up tests to do individual tests for each group of methds so we can (fixme)
R: 43, 0: Too few public methods (0/2) (too-few-public-methods)

**Comment by KSchopmeyer** [comment]
Karl agreed to take test_wbemconnection_mock, run_cim_operations.py and Module pywbem_mock._wbemconnection_mock

**Comment by andy-maier** [comment]
Closing this one as resolved in 0.13.0. We'll have a new one for the next release.
